src/inc/til/env.h

```diff
@@ -0,0 +1,575 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include <wil/token_helpers.h>
+#include <winternl.h>
+
+#ifdef UNIT_TESTING
+class EnvTests;
+#endif
+
+namespace til // Terminal Implementation Library. Also: "Today I Learned"
+{
+    namespace details
+    {
+
+        //
+        // A case-insensitive wide-character map is used to store environment variables
+        // due to documented requirements:
+        //
+        //      "All strings in the environment block must be sorted alphabetically by name.
+        //      The sort is case-insensitive, Unicode order, without regard to locale.
+        //      Because the equal sign is a separator, it must not be used in the name of
+        //      an environment variable."
+        //      https://docs.microsoft.com/en-us/windows/desktop/ProcThread/changing-environment-variables
+        //
+        // - Returns CSTR_LESS_THAN, CSTR_EQUAL or CSTR_GREATER_THAN
+        [[nodiscard]] int compare_string_ordinal(const std::wstring& lhs, const std::wstring& rhs) noexcept

```

Comment:
```suggestion
        [[nodiscard]] int compare_string_ordinal(const std::wstring_vew& lhs, const std::wstring_vew& rhs) noexcept
```

...otherwise it'll unnecessarily create new string instances when calling this function in `check_for_temp` or `is_path_var`.